.github/workflows/typing.yml

```diff
@@ -0,0 +1,63 @@
+# Workflow to generate and test type annotations
+name: Typing
```

Comment:
@stefanv, I know you suggested to use a more specific name here. But currently I truly intend this workflow to handle everything around typing: stub creation and stub testing with MyPy, etc. And long-term this might even deal with inlined type annotations once docstub supports it. 

And given the large number of workflows listed in the [repo's action tab](https://github.com/scikit-image/scikit-image/actions) I'd rather avoid creating even more entries with obsolete names there.